internal/backend/azure/azure.go

```diff
@@ -139,84 +156,19 @@ func (be *Backend) Save(ctx context.Context, h restic.Handle, rd restic.RewindRe
 
 	be.sem.GetToken()
 
-	debug.Log("InsertObject(%v, %v)", be.container.Name, objName)
-
-	var err error
-	if rd.Length() < 256*1024*1024 {
-		// wrap the reader so that net/http client cannot close the reader
-		// CreateBlockBlobFromReader reads length from `Len()``
-		dataReader := azureAdapter{rd}
+	debug.Log("InsertObject(%v, %v)", be.container, objName)
 
-		// if it's smaller than 256miB, then just create the file directly from the reader
-		err = be.container.GetBlobReference(objName).CreateBlockBlobFromReader(dataReader, nil)
-	} else {
-		// otherwise use the more complicated method
-		err = be.saveLarge(ctx, objName, rd)
+	// if it's smaller than 256miB, then just create the file directly from the reader
+	blob := be.container.NewBlockBlobURL(objName)
 
-	}
+	_, err := storage.UploadStreamToBlockBlob(context.TODO(), rd, blob, storage.UploadStreamToBlockBlobOptions{})
```

Comment:
I'd suggest to use a larger buffer size than the default 1MB. The old code used 100MB buffers, so maybe something inbetween would be optimal?

Reply:
Also, shouldn't this use ctx?